pkg/machinery/constants/constants.go

```diff
@@ -232,8 +232,23 @@ const (
 	// KubernetesSchedulerSecretsDir defines ephemeral directory with kube-scheduler secrets.
 	KubernetesSchedulerSecretsDir = KubebernetesStaticSecretsDir + "/" + "kube-scheduler"
 
-	// KubernetesRunUser defines UID to run control plane components.
-	KubernetesRunUser = 65534
+	// KubernetesAPIServerRunUser defines UID to the API Server.
+	KubernetesAPIServerRunUser = 65534
+
+	// KubernetesAPIServerRunGroup defines GID to run the API Server.
+	KubernetesAPIServerRunGroup = 65534
+
+	// KubernetesControllerManagerRunUser defines UID to the Controller Manager.
+	KubernetesControllerManagerRunUser = 65535
+
+	// KubernetesControllerManagerRunGroup defines GID to run the Controller Manager.
+	KubernetesControllerManagerRunGroup = 65535
```

Comment:
any reason why everything just can run as 65535. I guess the security worry is that different k8s api components could potentially access stuff owned by other component

Reply:
Indeed, it add an extra layer of defense. Practically this protection layers acts in two fold:
1) it protects against containerd/runc/kubelet( for kubelet would not be a first!) security mount bugs which allows for more file access on the host.
2) it adds an extra layer of defense as a container escape would not be enough the extract more secrets (because of the filesystem owner). This is because you are still the same user on the host when you escape a container.

In both cases the process userid is still evaluated when access files.

Reply:
And a less important argument: it follows the KEP (kubeadm is doing it too)